We leave configs in /etc/mysql only for humans.
As /etc/mysql/my.cnf is mediated and can point to
another DBMS's config, we don't read it.
We also skip /etc/my.cnf as we are not sure that it's
our own config. So, we use only $(CONFDIR)/my.cnf

--- percona-server-5.5.50-38.0/mysys/default.c.~1~	2016-06-30 13:04:53.000000000 +0300
+++ percona-server-5.5.50-38.0/mysys/default.c	2016-07-27 15:39:27.231412249 +0300
@@ -1226,12 +1226,12 @@
 
 #else
 
-  errors += add_directory(alloc, "/etc/", dirs);
-  errors += add_directory(alloc, "/etc/mysql/", dirs);
-
 #if defined(DEFAULT_SYSCONFDIR)
   if (DEFAULT_SYSCONFDIR[0])
     errors += add_directory(alloc, DEFAULT_SYSCONFDIR, dirs);
+#else
+  errors += add_directory(alloc, "/etc/", dirs);
+  errors += add_directory(alloc, "/etc/mysql/", dirs); 
 #endif /* DEFAULT_SYSCONFDIR */
 
 #endif
